source/connection-monitoring-and-pooling/tests/pool-checkout-returned-connection-maxConnecting.yml

```diff
@@ -0,0 +1,106 @@
+version: 1
+style: integration
+description: threads blocked by maxConnecting check out returned connections
+runOn:
+  -
+    # required for blockConnection in fail point
+    minServerVersion: "4.4.0"
+failPoint:
+  configureFailPoint: failCommand
+  mode: { times: 3 }
+  data:
+    failCommands: ["isMaster"]
+    closeConnection: false
+    blockConnection: true
+    blockTimeMS: 750
+poolOptions:
+  maxPoolSize: 10
+  waitQueueTimeoutMS: 5000
+operations:
```

Comment:
This test fails for me (even with `times: 10`). Could we simplify this entire test by:
- create one connection on the main thread, `conn0`
- start 3 threads that all perform a single `checkOut`
- Wait for 3 `ConnectionCheckOutStarted` events on the main thread 
- checkin `conn0` on the main thread
- Wait for all 3 threads to complete
- Assert that only 3 total connections were created, not 4.

Reply:
Done, although I don't think I needed to change the event assertions to match this scheme.

Re: failure with high `times`, should I set this to `alwaysOn` as well?

Reply:
high-but-finite value is safer than alwaysOn imo.

Reply:
increased to 10. I imagine once we hash out the fairness issue in the other thread, 10 should be high enough to pass this test too.